src/NzbDrone.Core/Datastore/Migration/023_add_release_groups_etc.cs

```diff
@@ -0,0 +1,231 @@
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+using NzbDrone.Common.Serializer;
+using System.Collections.Generic;
+using NzbDrone.Core.Music;
+using System.Data;
+using System;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(023)]
+    public class add_release_groups_etc : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            // ARTISTS TABLE
+            
+            Create.TableForModel("ArtistMetadata")
+                .WithColumn("ForeignArtistId").AsString().Unique()
+                .WithColumn("Name").AsString()
+                .WithColumn("Overview").AsString().Nullable()
+                .WithColumn("Disambiguation").AsString().Nullable()
+                .WithColumn("Type").AsString().Nullable()
+                .WithColumn("Status").AsInt32()
+                .WithColumn("Images").AsString()
+                .WithColumn("Links").AsString().Nullable()
+                .WithColumn("Genres").AsString().Nullable()
+                .WithColumn("Ratings").AsString().Nullable()
+                .WithColumn("Members").AsString().Nullable();
+
+            // we want to preserve the artist ID.  Shove all the metadata into the metadata table.
+            Execute.Sql(@"INSERT INTO ArtistMetadata (ForeignArtistId, Name, Overview, Disambiguation, Type, Status, Images, Links, Genres, Ratings, Members)
+                          SELECT ForeignArtistId, Name, Overview, Disambiguation, ArtistType, Status, Images, Links, Genres, Ratings, Members
+                          FROM Artists");
+            
+            // Add an ArtistMetadataId column to Artists
+            Alter.Table("Artists").AddColumn("ArtistMetadataId").AsInt32().WithDefaultValue(0);
+
+            // Update artistmetadataId
+            Execute.Sql(@"UPDATE Artists
+                          SET ArtistMetadataId = (SELECT ArtistMetadata.Id 
+                                                  FROM ArtistMetadata 
+                                                  WHERE ArtistMetadata.ForeignArtistId = Artists.ForeignArtistId)");
+
+            // RELEASES TABLE - Do this before we mess with the Albums table
+
+            Create.TableForModel("Releases")
```

Comment:
This table name could possibly get confused with indexer releases, to which we have an API endpoint module and resource named similar. Should we use AlbumReleases?

Reply:
I've renamed to Album Release locally. Afraid I got distracted by the parsing we were talking about so forgot to push. Will do it tomorrow.